contracts/sktl.sol

```diff
@@ -117,36 +117,36 @@ contract SKTL is
             // minting
             return;
 
-        uint256 rewardTokenTransfered = (value * totalRewardToken) /
-            totalSupply();
+        // uint256 rewardTokenTransfered = (value * totalRewardToken) /
+        //     totalSupply();
 
-        _rewardTokenBalance[from] -= rewardTokenTransfered;
-        _rewardTokenBalance[to] += rewardTokenTransfered;
+        // _rewardTokenBalance[from] -= rewardTokenTransfered;
+        // _rewardTokenBalance[to] += rewardTokenTransfered;
     }
 
     function increaseReward(uint256 amount) public onlyOwner {
         _mint(owner(), amount);
 
         // scale the deposit and add the previous remainder
         uint256 scaledAvailable = (amount * scaling) + _scaledRemainder;
-        _scaledRewardPerRewardToken += scaledAvailable / totalRewardToken;
-        _scaledRemainder = scaledAvailable % totalRewardToken;
-        _scaledRewardCreditedTo[owner()] = _scaledRewardPerRewardToken;
+        _scaledRewardPerToken += scaledAvailable / totalSupply();
```

Comment:
Are you sure you want to use totalSupply() here?  Wouldn't that include the supply of tokens you just minted as the reward?  Maybe you should mint the reward tokens after calculating the scaled reward per existing token to avoid possibly including them in the total supply amount which could throw off the reward amount (you don't reward bonus tokens for the bonus tokens you are issuing)